test/test_boot.py

```diff
@@ -24,6 +30,27 @@ def test_boot(self):
             "-device", "virtserialport,chardev=stdio",
 
             f"{self.get_path_to_store(output_id)}/f30-boot.qcow2"
-        ], encoding="utf-8", stdout=subprocess.PIPE, check=True)
-
-        self.assertEqual(r.stdout.strip(), "running")
+        ]
+
+        # Run qemu.
+        print("Booting image via qemu.")
+        r = subprocess.Popen(
+            qemu_boot_command,
+            encoding="utf-8",
+            stdout=subprocess.PIPE
+        )
+
+        # Monitor the qemu output for the "running" string which shows when
+        # the instance has fully booted.
+        print("Waiting for instance to fully boot.")
+        for line in iter(r.stdout.readline, ''):
+            # If the VM writes "running" to the console, everything worked!
+            if "running" in line:
+                print("Instance has fully booted!")
+                boot_successful = True
+                break
+
+        # Stop qemu if it is still running.
+        r.terminate()
+
+        self.assertTrue(boot_successful)
```

Comment:
Can you elaborate on this change?

You change the code from checking for equality of the output to just verifying that *some* output-line matches the expected output. Why is that? Is qemu printing debug information? Or do you want to be able to easily add debug information to qemu-output without causing the test to fail?

Secondly, you no longer wait for it to gracefully shutdown, but rather use an explicit termination. This feels weird because we no longer verify that the shutdown was successful. However, I guess we never did that because the return-code of qemu is always 0, since it has no feature of detecting whether the machine it run shutdown gracefully. So I guess that change is fine, or is there more to it?